### DISCUSSION TITLE
Best way to toggle specific polybar?

### DISCUSSION THREAD

**Discussion by joshpetit** [polybar/polybar#2780] [Q&A] [answered]
So I have 3 different polybars running, and I would like to toggle one of them. I know polybar-msg takes in the process tag, but trying to figure out which process the specific bar is is a little difficult. Has anyone been able to achieve this in a straightforward way?

For clarity this is what I would like to have:

```
polybar-msg --bar bottom-bar cmd toggle
```

And this would only toggle the bottom bar and not the top bar. Does a way to achieve this functionality exist?

**Comment by joshpetit** [comment, marked as answer]
I got it done using xdotool:

```
org_bar=$(xdotool search --name polybar-bottom getwindowpid)
```
Where `polybar-bottom` is the beginning of the name of the bar that I want to toggle. With this process id I do

```
polybar-msg -p $org_bar cmd toggle
```

**Comment by vtmx** [comment]
I created this effect in a script:

```
if [[ -z $(pgrep -a polybar) ]]; then
  polybar
else
  killall polybar
  while pgrep polybar >/dev/null; do sleep 1; done
fi  
```
